src/sql/workbench/contrib/notebook/browser/notebookViews/notebookViewsCodeCell.component.ts

```diff
@@ -4,26 +4,87 @@
+import { CellModel } from 'sql/workbench/services/notebook/browser/models/cell';
 import { localize } from 'vs/nls';
 
-
 export const CODE_SELECTOR: string = 'views-code-cell-component';
 
 @Component({
 	selector: CODE_SELECTOR,
+	changeDetection: ChangeDetectionStrategy.OnPush,
 	templateUrl: decodeURI(require.toUrl('./notebookViewsCodeCell.component.html'))
 })
 
-export class NotebookViewsCodeCellComponent extends CodeCellComponent {
-	public readonly emptyCellText: string = localize('viewsCodeCell.emptyCellText', "Please run this cell to view outputs.");
+export class NotebookViewsCodeCellComponent extends CodeCellComponent implements OnChanges {
+	@Input() display: boolean;
+	@Input() modal: boolean;
+
+	public override stdIn: nb.IStdinMessage;
 
 	constructor(@Inject(forwardRef(() => ChangeDetectorRef)) changeRef: ChangeDetectorRef) {
 		super(changeRef);
 	}
 
-	get outputs(): nb.ICellOutput[] {
-		return this.cellModel.outputs.filter((output: nb.IDisplayResult) => output.data && output.data['text/plain'] !== '<IPython.core.display.HTML object>');
+	override ngOnInit() {
+		if (this.cellModel) {
+			this._register(this.cellModel.onCollapseStateChanged((state) => {
+				this._changeRef.detectChanges();
+			}));
+			this._register(this.cellModel.onParameterStateChanged((state) => {
+				this._changeRef.detectChanges();
+			}));
+			this._register(this.cellModel.onOutputsChanged(() => {
+				this._changeRef.detectChanges();
+			}));
+
+			// If we have a pre-existing message, listen to that
+			if (this.cellModel?.future?.msg) {
+				this.handleStdIn(this.cellModel.future.msg as nb.IStdinMessage);
+			}
+
+			// Register request handler, cleanup on dispose of this component
+			this.cellModel.setStdInHandler({ handle: (msg) => super.handleStdIn(msg) });
+			this._register({ dispose: () => this.cellModel.setStdInHandler(undefined) });
+		}
+	}
+
+	override ngOnChanges(changes: { [propKey: string]: SimpleChange }) {
+		for (let propName in changes) {
+			if (propName === 'activeCellId') {
+				let changedProp = changes[propName];
+				super._activeCellId = changedProp.currentValue;
+				break;
+			}
+		}
+	}
+
+	get outputs(): readonly nb.ICellOutput[] {
+		return this.cellModel.outputs;
+	}
+
+	/**
+	 * Override the cell model for Views.
+	 */
+	get viewCellModel(): ICellModel {
+		return new NotebookViewsCellModel(this.cellModel.toJSON(), { notebook: this.cellModel.notebookModel, isTrusted: this.cellModel.trustedMode });
+	}
+
+	get emptyCellText(): string {
+		return localize('viewsCodeCell.emptyCellText', "Please run this cell to view outputs.");
+	}
+}
+
+export class NotebookViewsCellModel extends CellModel {
+	/**
+	 * Override the cell output.
+	 */
+	public override get outputs(): Array<nb.ICellOutput> {
+		return super.outputs
+			.filter((output: nb.IDisplayResult) => output.data === undefined || output?.data['text/plain'] !== '<IPython.core.display.HTML object>')
+			.map((output: nb.ICellOutput) => ({ ...output }))
```

Comment:
Would suggest adding a comment here what these multiple mappings are supposed to accomplish